ReactWindows/ReactNative/Views/TextInput/ReactTextInputManager.cs

```diff
@@ -547,6 +548,12 @@ public override void ReceiveCommand(ReactTextBox view, int commandId, JArray arg
             if (commandId == FocusTextInput)
             {
                 view.Focus(FocusState.Programmatic);
+                
+                // Open the touch keyboard if the device is in touch mode
+                if (UIViewSettings.GetForCurrentView().UserInteractionMode == UserInteractionMode.Touch)
```

Comment:
This sounds like the exact property that I wanted to achieve and after looking it up on the [Microsoft API reference](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.preventkeyboarddisplayonprogrammaticfocus), it should also "automatically detect" whether or not it should show the onscreen keyboard (last input by touch?).

Unfortunately, when I tried this out with the RNTester, the keyboard was never automatically displayed for the auto-focus input. I exposed the property to RN and also made sure that it is actually set. The field is focused, but the keyboard stays hidden when the page opens.

Any idea why this might be? I tested it on the Surface Pro with no keyboard attached.

Reply:
From what I recall, provided that there isn't physical keyboard attached, xaml should detect and on focus deploy the on-screen keyboard. I would suggest 2 things:
1. Test the behavior on a standalone application, without React Native - just a play text input and see if it works there.
2. If it works than try to execute the view.Focus command as Post on the UI thread. Focus sometimes does odd things and needs an extra loop to work properly


---
In reply to: [238218220](https://github.com/Microsoft/react-native-windows/pull/2029#discussion_r238218220) [](ancestors = 238218220)

Reply:
Sorry for the delay.

I created a standalone UWP application with a single TextBox which gets focused by a timer after a few seconds. I then ran this app on my Surface Pro with no keyboard attached and the latest input being a tap on the touch screen as mentioned on the [API reference](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.preventkeyboarddisplayonprogrammaticfocus). The TextBox gets the focus, but the keyboard stays hidden.

I also tried to make sure `Focus()` is called on the UI thread by using a Dispatcher, but this doesn't change the result. I did all of this with the RNTester as well with the same outcome. Do you have an idea why that might be or what else I could try?

Given the outcome so far, I think using `InputPane.GetForCurrentView().TryShow()` might be the only approach to show the touch keyboard as desired in touch mode.